src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -863,30 +863,23 @@ private void CreatePipe(out SafeFileHandle parentHandle, out SafeFileHandle chil
 
         private static string GetEnvironmentVariablesBlock(IDictionary<string, string> sd)
         {
-            // get the keys
-            string[] keys = new string[sd.Count];
-            sd.Keys.CopyTo(keys, 0);
-
-            // sort both by the keys
-            // Windows 2000 requires the environment block to be sorted by the key
-            // It will first converting the case the strings and do ordinal comparison.
+            // https://docs.microsoft.com/en-us/windows/win32/procthread/changing-environment-variables
+            // "All strings in the environment block must be sorted alphabetically by name. The sort is
+            //  case-insensitive, Unicode order, without regard to locale. Because the equal sign is a
+            //  separator, it must not be used in the name of an environment variable."
 
-            // We do not use Array.Sort(keys, values, IComparer) since it is only supported
-            // in System.Runtime contract from 4.20.0.0 and Test.Net depends on System.Runtime 4.0.10.0
-            // we workaround this by sorting only the keys and then lookup the values form the keys.
+            var keys = new string[sd.Count];
+            sd.Keys.CopyTo(keys, 0);
             Array.Sort(keys, StringComparer.OrdinalIgnoreCase);
 
-            // create a list of null terminated "key=val" strings
-            StringBuilder stringBuff = new StringBuilder();
-            for (int i = 0; i < sd.Count; ++i)
+            // Join the null-terminated "key=val\0" strings
+            var result = new StringBuilder();
```

Comment:
```suggestion
            var result = new StringBuilder(8 * keys.Length);
```
Would it be worth it to pre-allocate some space based on `keys.Length`?

Reply:
Sure.  I'm not sure how much is worth preallocating... on my machine, the average space needed per key/value pair would actually be 54.  But we could start with 8 as you suggested, which is almost certainly going to be too small but better than the default.

Reply:
Yeah, it will just about always be too small, I had misread this as 80.

Reply:
@stephentoub on my machine, the average is about 37 chars per key in a regular prompt, and 59 in a VS prompt. Why not use a value of 40-80? A value of 8 is guaranteed to resize.

I guess it doesn't matter as it's small beans compared to a process start, but it just seems wierd.

Reply:
8 would be appropriate if the average env var being passed in was `FOO=BAR\0` or shorter..

Reply:
StringBuilder does not resize. It maintains linked list of buffers. `8` allows us to start with larger buffer in the linked list (ie reduce number of buffers in the linked list). Being more aggresive would reduce the number of buffers in the linked list even more, but it has a risk of overestimating the size.

Reply:
Fair enough